CenterForOpenScience/ember-osf-web
# [ENG-3571] Require unregistered user email

-   Ticket: [ENG-3571]
-   Feature flag: n/a

## Purpose
- Make unregistered user email a required field in the Frontend GUI

## Summary of Changes
- Disallow blank emails and names for unregistered users
- Explicitly require email field
- Update tests

## Screenshot(s)
- Before:

-After:
![image](https://user-images.githubusercontent.com/51409893/187279076-7d3adaec-64a9-4c29-88a2-cccd02a20e74.png)
![image](https://user-images.githubusercontent.com/51409893/187280782-74bb5cd6-2cbe-45d3-ad19-7a96a658148e.png)


## Side Effects
- None

## QA Notes
- Please check that the validation works as expected and that the Add button is disabled properly when the form is invalid

[ENG-3571]: https://openscience.atlassian.net/browse/ENG-3571?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
